authenticity-collector/src/identity/pipes/identity-validation.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { PipeTransform, Injectable, ArgumentMetadata, BadRequestException } from '@nestjs/common';
+import { validate } from 'class-validator';
+import { plainToInstance } from 'class-transformer';
+import { IdentityDto } from '../models/IdentityDto';
+
+@Injectable()
+export class IdentityValidationPipe implements PipeTransform<IdentityDto> {
+    async transform(value: any, { metatype }: ArgumentMetadata) {
+        if (!metatype || !this.toValidate(metatype)) {
+            return value;
+        }
+        value.timestamp = new Date(value.timestamp);
```

Comment:
Is this line even needed? 

If needed then:
please do not edit input parameters it would be better to create a new local var here as following:

```
const v = {...value, timestamp:new Date(value.timestamp)}
and use v in the following statements
```

If it is not needed please remove this line